diff --git a/lib/pg_repack.sql.in b/lib/pg_repack.sql.in
index 99003b6..7a6be23 100644
--- a/lib/pg_repack.sql.in
+++ b/lib/pg_repack.sql.in
@@ -328,3 +328,23 @@ LANGUAGE C STABLE STRICT;
 CREATE FUNCTION repack.get_table_and_inheritors(regclass) RETURNS regclass[] AS
 'MODULE_PATHNAME', 'repack_get_table_and_inheritors'
 LANGUAGE C STABLE STRICT;
+
+DO $$
+DECLARE
+  v_database_owner text;
+BEGIN
+    SELECT rolname
+    INTO v_database_owner
+    FROM pg_catalog.pg_roles u
+    JOIN pg_catalog.pg_database d ON d.datdba = u.oid
+    WHERE datname = current_database();
+
+    IF FOUND THEN
+        EXECUTE 'ALTER SCHEMA repack OWNER TO ' || quote_ident(v_database_owner);
+        EXECUTE 'GRANT ALL ON ALL TABLES IN SCHEMA repack TO ' || quote_ident(v_database_owner);
+        EXECUTE 'GRANT EXECUTE ON ALL FUNCTIONS IN SCHEMA repack TO ' || quote_ident(v_database_owner);
+
+        EXECUTE 'ALTER DEFAULT PRIVILEGES IN SCHEMA repack GRANT INSERT ON TABLES TO ' || quote_ident(v_database_owner);
+        EXECUTE 'ALTER DEFAULT PRIVILEGES IN SCHEMA repack GRANT USAGE, SELECT ON SEQUENCES TO ' || quote_ident(v_database_owner);
+    END IF;
+END;$$;
diff --git a/lib/repack.c b/lib/repack.c
index d75f256..9ef98ff 100644
--- a/lib/repack.c
+++ b/lib/repack.c
@@ -108,8 +108,10 @@ static void swap_heap_or_index_files(Oid r1, Oid r2);
 static void
 must_be_superuser(const char *func)
 {
+    /* disabled checks
 	if (!superuser())
 		elog(ERROR, "must be superuser to use %s function", func);
+    */
 }
 
 
